File: Chapter 04/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter04/section_4_4_anti_patterns/before_fat_controller_anemic_domain/SmtpEmailService.java
package com.grokkingsoftwarearchitecture.chapter04.section_4_4_anti_patterns.before_fat_controller_anemic_domain;

import com.grokkingsoftwarearchitecture.chapter04.shared.LogManager;

/**
 * INFRASTRUCTURE LAYER: EXTERNAL SERVICE.
 * ARCHITECTURE NOTE: Directly instantiating an SMTP service 
 * inside a Controller makes the code slow and fragile.
 */
public class SmtpEmailService {
    public void send(String email, String message) {
        LogManager.info(SmtpEmailService.class, "  [Email] SMTP Logic: Sending '{0}' to {1}",
            message, email);
    }
}

File: Chapter 04/Python/section_4_4_anti_patterns/before_fat_controller_anemic_domain/smtp_email_service.py
"""
INFRASTRUCTURE LAYER: EMAIL SERVICE.
ARCHITECTURE NOTE: Tightly coupling this service to the 
Controller makes unit testing impossible.
"""
class SmtpEmailService:
    def send(self, email: str, message: str):
        print(f"  [Email] SMTP: Sending '{message}' to {email}")